ciml/gather_results.py

```diff
@@ -587,6 +587,82 @@ def save_dataset(dataset, name, data_path=None, s3=None, **kwargs):
+@click.option('--data-keys', '-k', multiple=True, help="Key to export")
+@click.option('--data-path', default=None,
+              help="Path to the raw data, local path or s3://<bucket>")
+@click.option('--s3-profile', default='ibmcloud', help='Named configuration')
+@click.option('--s3-url',
+              default='https://s3.eu-geo.objectstorage.softlayer.net',
+              help='Endpoint URL for the s3 storage')
+@click.option('--output', help="Name of the output file")
+@click.option('--title', help="Title of the graph")
+def plot_experiment_data(dataset_experiment_label,
+                         dataset_experiment_comp, experiment_sets_names,
+                         data_keys, data_path,
+                         s3_profile, s3_url, output, title):
+    # Do some input validation when dataset_experiment_label_comp is set
+    if dataset_experiment_comp:
+        if not experiment_sets_names:
+            print("Please provide the name for the two experiment sets")
+            sys.exit(1)
+        d1_len = len(dataset_experiment_label)
+        d2_len = len(dataset_experiment_comp)
+        if d1_len != d2_len:
+            print("The experiments to compare must have the same size")
+            print("Dataset1: %d, dataset2 %d" % (d1_len, d2_len))
+            sys.exit(1)
+        if len(data_keys) > 1:
+            print("Cannot have two experiments and multiple keys")
+            sys.exit(1)
+
+    s3 = get_s3_client(s3_profile=s3_profile, s3_url=s3_url)
+    # Define the array size
+    array_size = 2 if dataset_experiment_comp else len(data_keys)
+    data = np.ndarray(shape=(len(dataset_experiment_label), array_size))
+    labels = []
+    for count, d_e_l in enumerate(dataset_experiment_label):
+        exp_data = load_data_json(d_e_l[0], 'eval_' + d_e_l[0],
+                                  sub_folder=d_e_l[1],
+                                  data_path=data_path, s3=s3)
+        labels.append(d_e_l[2])
+        if dataset_experiment_comp:
+            d2, e2 = dataset_experiment_comp[count]
+            exp_data_comp = load_data_json(d2, 'eval_' + d2,
+                                           sub_folder=e2,
+                                           data_path=data_path, s3=s3)
+        data_count = []
+        for count_k, k in enumerate(data_keys):
+            if k == 'accuracy':
+                data_count.append(1 - exp_data[k])
+                if dataset_experiment_comp:
+                    data_count.append(1 - exp_data_comp[k])
+            else:
+                data_count.append(exp_data[k])
+                if dataset_experiment_comp:
+                    data_count.append(exp_data_comp[k])
+
+        data[count] = data_count
+
+    # Setup the dataframe
+    if not dataset_experiment_comp:
+        columns = data_keys
+    else:
+        columns = experiment_sets_names
+    df = pd.DataFrame(data, index=labels, columns=columns)
+    print("Plotting %s with data %s" % (title, df))
```

Comment:
This feels a bit redundant to me (since we can just look at the plot) but I guess it doesn't hurt.